### PR TITLE
Add Spanner to the enforce_labels constraint.

### PR DESCRIPTION
**Pull request by mmontan**
Spanner CAI and API formats are different, and the data generated by Terraform Validator requires customizing the label enforce check.

Related PR in Validator:
https://github.com/GoogleCloudPlatform/terraform-validator/pull/137

### CODE REVIEW

File: validator/enforce_labels.rego [hunk 1]
```diff
@@ -91,6 +92,35 @@ get_labels(asset, non_standard_types) = resource_labels {
 	resource_labels := lib.get_default(resource, "resourceLabels", {})
 }
 
+# get_labels for Spanner Instances. API format.
+# This method matches data generated by Terraform Validator and other
+# tools that use the API format to represent resources.
+get_labels(asset, non_standard_types) = resource_labels {
+	# check if we have a non-standard type
+	asset.asset_type == non_standard_types[_]
```
Comment (morgante) [inline validator/enforce_labels.rego:100]: Note, this line is redundant/unnecessary.

File: validator/enforce_labels.rego [hunk 2]
```diff
@@ -91,6 +92,35 @@ get_labels(asset, non_standard_types) = resource_labels {
...
+	resource := asset.resource.data.instance
+	resource_labels := lib.get_default(resource, "labels", {})
+}
+
+# get_labels for Spanner Instances. CAI format.
+# This method matches data generated by CAI, where the API
+# format has been simplified.
+get_labels(asset, non_standard_types) = resource_labels {
+	# check if we have a non-standard type
+	asset.asset_type == non_standard_types[_]
```
Comment (morgante) [inline validator/enforce_labels.rego:114]: Note this line is unnecessary.

File: validator/enforce_labels_test.rego [hunk 3]
```diff
@@ -414,3 +423,27 @@ test_enforce_label_bq_violations {
 	is_string(violation.msg)
 	is_object(violation.details)
 }
+
+#### Testing for Spanner Instances
+# Confirm 2 violation, one for the CAI representation and one for the API
+test_enforce_label_spanner_violations {
```
Comment (morgante) [inline validator/enforce_labels_test.rego:429]: Could we switch this to use the new test utils? https://github.com/forseti-security/policy-library/blob/master/validator/test_utils.rego

Comment (mmontan) [reply validator/enforce_labels_test.rego:263]: Updated all tests to use the new library.

File: validator/enforce_labels.rego (hunk 2 above)
Comment (mmontan) [reply validator/enforce_labels.rego]: Removed

File: validator/enforce_labels.rego (hunk 1 above)
Comment (mmontan) [reply validator/enforce_labels.rego]: Removed

**Pull request merged**